src/test/java/de/bonndan/nivio/input/GroupQueryResolverTest.java

```diff
@@ -0,0 +1,97 @@
+package de.bonndan.nivio.input;
+
+import de.bonndan.nivio.input.dto.GroupDescription;
+import de.bonndan.nivio.input.dto.LandscapeDescription;
+import de.bonndan.nivio.model.Group;
+import de.bonndan.nivio.model.Item;
+import de.bonndan.nivio.model.LandscapeImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class GroupQueryResolverTest {
+
+    private GroupQueryResolver groupResolver;
+
+    @Mock
+    private ProcessLog processLog;
+
+    @BeforeEach
+    public void setup() {
+        groupResolver = new GroupQueryResolver(processLog);
+    }
+
+    @Test
+    void process_doesNotContainDuplicatedItemsInGroups() {
+        LandscapeImpl landscape = new LandscapeImpl();
+        landscape.setProcessLog(processLog);
+        Item item = new Item();
+        item.setIdentifier("itemIdentifier");
+        item.setGroup("groupIdentifier");
+        landscape.setItems(Set.of(item));
+        landscape.addGroup(new Group("groupIdentifier"));
+
+        LandscapeDescription input = new LandscapeDescription();
+        GroupDescription groupDescription = new GroupDescription();
+        groupDescription.setIdentifier("groupIdentifier");
+        groupDescription.setContains(List.of("itemidentifier"));
+        input.getGroups().put("group", groupDescription);
+
+        groupResolver.process(input, landscape);
+
+        assertThat(landscape.getItems().stream().count()).isEqualTo(1L);
+        assertThat(landscape.getGroup("groupIdentifier").getItems()).containsExactly(item);
```

Comment:
This list contains the same `Item` twice. This is because the items are just added to the modifiable Collection in `Group`. 

I would refactor this to add a method `addItem` to `Group` that does not add items twice (or use a Set) and to have an unmodifiable Collection instead.

Reply:
As you suggested we are now using a Set.

Reply:
The unmodifiable Collection thing is a bit too much at the moment. We are constantly updating the collections in the indexing phase, and having side-effect free copies of copies of copies does not seem worth the effort to me.

Reply:
Ok, I guess most of my suggestions would require some other patterns in our code, which can be hard to implement now into the existing code base with it's patterns already in place.

Reply:
Yes. Please feel free to open tickets for enhancement proposals and simply extend #212